src/cli/self_update/windows.rs

```diff
@@ -809,110 +749,86 @@ pub(crate) fn delete_rustup_and_cargo_home(process: &Process) -> Result<()> {
 
 #[cfg(test)]
 mod tests {
-    use std::ffi::OsString;
-    use std::os::windows::ffi::OsStrExt;
-
-    use winreg::enums::{RegType, HKEY_CURRENT_USER, KEY_READ, KEY_WRITE};
-    use winreg::{RegKey, RegValue};
+    use windows_registry::CURRENT_USER;
+    use windows_result::HRESULT;
+    use windows_sys::Win32::Foundation::ERROR_FILE_NOT_FOUND;
 
     use crate::currentprocess::TestProcess;
     use crate::test::with_saved_path;
 
-    fn wide(str: &str) -> Vec<u16> {
-        OsString::from(str).encode_wide().collect()
-    }
-
     #[test]
     fn windows_install_does_not_add_path_twice() {
         assert_eq!(
             None,
             super::_add_to_path(
-                wide(r"c:\users\example\.cargo\bin;foo"),
-                wide(r"c:\users\example\.cargo\bin")
+                r"c:\users\example\.cargo\bin;foo",
+                r"c:\users\example\.cargo\bin"
             )
         );
     }
 
-    #[test]
-    fn windows_handle_non_unicode_path() {
-        let initial_path = vec![
-            0xD800, // leading surrogate
-            0x0101, // bogus trailing surrogate
-            0x0000, // null
-        ];
-        let cargo_home = wide(r"c:\users\example\.cargo\bin");
-        let final_path = [&cargo_home, &[b';' as u16][..], &initial_path].join(&[][..]);
-
-        assert_eq!(
-            &final_path,
-            &super::_add_to_path(initial_path.clone(), cargo_home.clone(),).unwrap()
-        );
-        assert_eq!(
-            &initial_path,
-            &super::_remove_from_path(final_path, cargo_home,).unwrap()
-        );
-    }
+    // FIXME: Must evaluate whether this test item should be removed.
```

Comment:
Should either remove this or fix it before merging I guess?

Reply:
I considered a while a decided to remove it, because using `windows-registry` will never appear non-unicode characters, I thought.